core/engine/src/builtins/temporal/plain_month_day/mod.rs

```diff
@@ -301,18 +314,45 @@ fn to_temporal_month_day(
     } else if let Some(item_string) = item.as_string() {
         InnerMonthDay::from_str(item_string.to_std_string_escaped().as_str())?
     } else if item.is_object() {
-        InnerMonthDay::new(
-            item.get_v(js_string!("month"), context)
-                .expect("Month not found")
-                .to_i32(context)
-                .expect("Cannot convert month to i32"),
-            item.get_v(js_string!("day"), context)
-                .expect("Day not found")
-                .to_i32(context)
-                .expect("Cannot convert day to i32"),
-            calendar,
-            overflow,
-        )?
+        let day = item
+            .get_v(js_string!("day"), context)
+            .expect("Day not found")
+            .to_i32(context)
+            .expect("Cannot convert day to i32");
+        let month = item
+            .get_v(js_string!("month"), context)
+            .expect("Month not found")
+            .to_i32(context)
+            .expect("Cannot convert month to i32");
+
+        let month_code = item
+            .get_v(js_string!("monthCode"), context)
+            .expect("monthCode not found");
+        let resolved_month_code = if month_code.is_undefined() {
+            None
+        } else {
+            TinyAsciiStr::<4>::from_str(
+                &month_code
+                    .to_string(context)
+                    .expect("Cannot convert monthCode to string")
+                    .to_std_string_escaped(),
+            )
+            .map_err(|e| JsError::from(JsNativeError::range().with_message(e.to_string())))
+            .ok()
+        };
+        let year = item.get_v(js_string!("year"), context).map_or(1972, |val| {
+            val.to_i32(context).expect("Cannot convert year to i32")
+        });
```

Comment:
Thought: this was already here, but I'm a bit worried that we're just `expect`ing a lot of missing values instead of throwing errors.

Reply:
😕 Hmmmm, that's a good point.